subprojects/core/src/main/java/org/gradle/internal/service/scopes/BasicGlobalScopeServices.java

```diff
@@ -88,16 +89,16 @@ DocumentationRegistry createDocumentationRegistry() {
         return new DocumentationRegistry();
     }
 
-    JvmMetadataDetector createJvmMetadataDetector(ExecHandleFactory execHandleFactory) {
-        return new CachingJvmMetadataDetector(new DefaultJvmMetadataDetector(execHandleFactory));
+    JvmMetadataDetector createJvmMetadataDetector(DefaultJvmMetadataDetector.Factory defaultJvmMetadataDetectorFactory) {
```

Comment:
Injecting a concrete implementation like that seems like a code smell. Do we need this factory at all?